src/app/(normal)/@modal/(.)post/[id]/photos/[index]/page.tsx

```diff
@@ -74,70 +39,5 @@ export default function PreviewPhotoModal({ params: { id: postId, index }, searc
-            <div
-                className="preview-actions fixed inset-0 flex transform-none flex-col items-center justify-center bg-black/90 bg-opacity-90 outline-none transition-all"
-                onClick={isMedium ? () => router.back() : undefined}
-            >
-                <div className="absolute left-4 top-4 cursor-pointer text-main">
-                    <CloseButton onClick={() => router.back()} />
-                </div>
-                <div className="flex w-full text-main">
-                    <Swiper
-                        modules={[Navigation, Keyboard]}
-                        navigation={{
-                            prevEl: prevRef.current,
-                            nextEl: nextRef.current,
-                        }}
-                        onBeforeInit={(swiper) => {
-                            if (typeof swiper.params.navigation === 'object') {
-                                swiper.params.navigation.prevEl = prevRef.current;
-                                swiper.params.navigation.nextEl = nextRef.current;
-                            }
-                        }}
-                        keyboard
-                        initialSlide={Number(index) - 1}
-                    >
-                        {images.map((asset, key) => {
-                            return (
-                                <SwiperSlide key={key} className="flex">
-                                    <div className="flex h-full w-full items-center justify-center">
-                                        <PreviewContent post={post!} asset={asset} />
-                                    </div>
-                                </SwiperSlide>
-                            );
-                        })}
-                        <ClickableButton
-                            ref={prevRef}
-                            className="prev-button absolute left-[50px] top-[50%] z-[9999] max-md:hidden"
-                        >
-                            <ArrowLeftIcon width={24} height={24} className="rounded-full p-1 text-main hover:bg-bg" />
-                        </ClickableButton>
-                        <ClickableButton
-                            ref={nextRef}
-                            className="next-button absolute right-[50px] top-[50%] z-[9999] max-md:hidden"
-                        >
-                            <ArrowRightIcon width={24} height={24} className="rounded-full p-1 text-main hover:bg-bg" />
-                        </ClickableButton>
-                    </Swiper>
-                </div>
-                <div className="absolute my-1 flex items-center justify-between bottom-safe">
-                    {post ? <PostActionsWithGrid className="gap-8" post={post} disablePadding /> : null}
-                </div>
-            </div>
-        </Modal>
-    );
+    return <PreviewMedia open post={post} source={currentSource} index={index} onClose={() => router.back()} />;
```

Comment:
Suggestion: separate `offset` and `index` from UI input, you can use `offset` as 0-indexed confidently inside `<PreviewMedia />`, besides, the UI output, don't forget to plus 1 to convert `offset` to `index`

```suggestion
    return <PreviewMedia open post={post} source={currentSource} offset={parse(index, 10)} onClose={() => router.back()} />;
```